R/auth.R

```diff
@@ -31,16 +32,13 @@ proof_authenticate <- function(username, password) {
   assert(username, "character")
   assert(password, "character")
 
-  response <- POST(make_url("authenticate"),
-    body = list(
+  request(make_url("authenticate")) |>
+    req_body_json(list(
       username = username,
       password = password
-    ),
-    encode = "json",
-    timeout(proofr_env$timeout_sec)
-  )
-  stop_for_status(response)
-  parsed <- content(response, as = "parsed")
-  token <- parsed$token
-  token
+    )) |>
+    req_timeout(proofr_env$timeout_sec) |>
+    req_perform() |>
+    resp_body_json() |>
+    {\(x) x$token}()
```

Comment:
Same thing here, if you want to use the native pipe you need to force R 4.1.0 or greater.